### DISCUSSION TITLE
Upgraded to Netbox 3.5... but only then realized I need this plugin. Migration doesn't work.

### DISCUSSION THREAD

**Discussion by Myke** [Onemind-Services-LLC/netbox-secrets#90] [Q&A] [answered]
I didn't know there was a 'new' secrets plugin for Netbox when I (... blindly) went ahead and did the 3.5 upgrade.

Unfortunately since the 'old' plugin refuses to play in Netbox 3.5, and the migration instructions on the 'new' plug-in page appear only update the SQL sequences... I'm not sure how to move my secrets forward.

Some advice please?

**Comment by kprince28** [comment]
When you run the migration then your old secrets will still be there.
And maybe this will be helpful to you
https://github.com/Onemind-Services-LLC/netbox-secrets/discussions/52

**Reply by Myke** [reply]
Thanks for the prompt reply, however the struggle is that the existing secrets themselves are what's not being migrated. 

The upgrade/migration directions in readme.md are assuming Netbox 3.4, which I've unfortunately moved past (and _really_ would prefer not to roll back). I see the new plugin in the GUI, but no secrets listed, with just a warning to create a role first.

I've done manage.py migrate and collectstatic several times, with various incantations of the plugins in the netbox configuration.py and local_requirements, however the outcome is the same: manage.py migrate.py errors our when the old plugin is referenced "Plugin netbox_secretstore requires NetBox maximum version 3.4.99." - and with or without it, the secrets aren't coming into the new plugin. 

UTSL is a bit above my head, but I'm comfortable wrangling Postgres & pgsql.

NB: Running Netbox and netbox-secrets (and others) from FreeBSD packages.

**Reply by abhi1693** [reply, marked as answer]
Well if you don't want to roll back then there is nothing we can do for you. Secretstore plugin was never released for v3.5 and as such we don't want to take the burden of providing a migration for every release in the future.

So currently, you'll have to either build all your secrets manually or you can take reference from the migration files to figure out a way via django shell.
